### PR TITLE
Dropdown component refactored, tests added

### LINKED ISSUE #1359: Refactors dropdown component and add tests
**I'm submitting a ...**
- [ ] bug report
- [ ] feature request
- [x] Refactoring

**Current behavior:**

The following issues persist in the dropdown component.
- Standard coding practice is not followed.
- Remove redundant CSS classes
- Increase coverage
- Minor UI fixes

**Will work on the issue:**
Yes

### PR DESCRIPTION
**Pull request by rajvaibhavdubey**
Fixes #1359

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)
- [x] Added Surge preview link

#### Changes proposed in this pull request:

- Dropdown Component has been refactored
- Minor UI fixes
- Also tests have been added for the basic skeleton of the dropdown component.
- Badgemagic has also been added to the dropdown

- https://pr-1367-fossasia-susper.surge.sh

### CODE REVIEW

File: src/app/dropdown/dropdown.component.spec.ts [hunk 1]
```diff
@@ -23,7 +23,21 @@ describe('DropdownComponent', () => {
...
   });
 
-  it('should have susperUrl variable', () => {
-    expect(component.susperUrl).toBeTruthy();
+  it('should have dropdown-menu', () => {
+    const compiled = fixture.debugElement.nativeElement;
+    expect(compiled.querySelector('div .side-menu')).toBeTruthy();
   });
+
+
```
Comment (sakshee-19) [inline src/app/dropdown/dropdown.component.spec.ts]: Remove extra lines, please.

File: src/app/dropdown/dropdown.component.spec.ts [hunk 2]
```diff
@@ -23,7 +23,21 @@ describe('DropdownComponent', () => {
...
+    const compiled = fixture.debugElement.nativeElement;
+    expect(compiled.querySelector('div .dropdown-menu')).toBeTruthy();
+  });
+
+  it('should have menu-row items', () => {
+    const compiled = fixture.debugElement.nativeElement;
+    expect(compiled.querySelector('div .menu-row')).toBeTruthy();
+  });
+
+
```
Comment (sakshee-19) [inline src/app/dropdown/dropdown.component.spec.ts]: Remove extra lines, please.

**Comment by shreyanshdwivedi** [comment]
@rajvaibhavdubey any update on this?

**Comment by rajvaibhavdubey** [comment]
@praveenojha33 Please review this. 
ThankYou :smile:

**Pull request merged**